azure/services/virtualmachines/virtualmachines.go

```diff
@@ -193,6 +200,14 @@ func (s *Service) Reconcile(ctx context.Context) error {
 			}
 		}
 
+		for _, dataDisk := range vmSpec.DataDisks {
+			if dataDisk.ManagedDisk != nil && dataDisk.ManagedDisk.StorageAccountType == UltraSSDStorageAccountType {
+				virtualMachine.VirtualMachineProperties.AdditionalCapabilities = &compute.AdditionalCapabilities{
+					UltraSSDEnabled: pointer.Bool(true),
```

Comment:
The rest of the file uses `to.*`
```suggestion
					UltraSSDEnabled: to.BoolPtr(true),
```